packages/marketplace/src/components/pages/admin-approvals/__tests__/admin-approvals.test.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  AdminApprovals,
+  handleCloseModal,
+  handleOnPageChange,
+  Content,
+  renderId,
+  renderViewDetailButton,
+  generateTableColumn,
+  HandleViewDetailOnClick,
+  handleViewDetailOnClick,
+} from '../admin-approvals'
+import Routes from '@/constants/routes'
+
+describe('admin-approvals', () => {
+  describe('handleCloseModal', () => {
+    it('should call dispatch', () => {
+      const setIsModalOpen = jest.fn()
+      const fn = handleCloseModal({ setIsModalOpen })
+      fn()
+      expect(setIsModalOpen).toBeCalled()
+    })
+  })
+
+  describe('handleOnPageChange', () => {
+    it('should call dispatch', () => {
+      const mockHistory = {
+        push: jest.fn(),
+      }
+      const fn = handleOnPageChange(mockHistory)
+      fn('1')
+      expect(mockHistory.push).toBeCalledWith(`${routes.ADMIN_APPROVALS}?page=1`)
+    })
+  })
+
+  describe('Content', () => {
+    const mockTableColumns = [
+      {
+        Header: 'AppId',
+        accessor: 'appId',
+      },
+      {
+        Header: 'Type',
+        accessor: 'type',
+      },
+      {
+        Header: 'Description',
+        accessor: 'description',
+      },
+    ]
+    it('should match snapshot', () => {
+      const wrapper = shallow(
+        <Content loading={false} waitingApprovalList={approvalsStub.data.data || []} tableColumns={mockTableColumns} />,
+      )
+      expect(wrapper).toMatchSnapshot()
+    })
+    it('should match snapshot', () => {
+      const wrapper = shallow(<Content loading={true} waitingApprovalList={[]} tableColumns={mockTableColumns} />)
+      expect(wrapper).toMatchSnapshot()
+    })
+  })
+
+  describe('renderId', () => {
+    it('should renderId correctly', () => {
+      const input = {
+        page: 1,
+      }
+      const wrapper = shallow(<div>{renderId(input)}</div>)
+      expect(wrapper).toMatchSnapshot()
+    })
+
+    it('should renderId correctly', () => {
+      const input = {
+        page: 2,
+      }
+      const wrapper = shallow(<div>{renderId(input)}</div>)
+      expect(wrapper).toMatchSnapshot()
+    })
+  })
+
+  describe('renderViewDetailButton', () => {
+    it('should render correctly', () => {
+      const mockProps = {
+        revisionDetail: revisionDetailDataStub.data,
+        appDetail: appDetailDataStub.data,
+        setIsModalOpen: jest.fn(),
+        dispatch: jest.fn(),
+      }
+      const wrapper = shallow(<div>{renderViewDetailButton(mockProps)}</div>)
+      expect(wrapper).toMatchSnapshot()
+    })
+  })
+
+  describe('generateTableColumn', () => {
+    it('should render correctly', () => {
+      const mockProps = {
+        page: 1,
+        revisionDetail: revisionDetailDataStub.data,
+        appDetail: appDetailDataStub.data,
+        setIsModalOpen: jest.fn(),
+        dispatch: jest.fn(),
+      }
+      const result = generateTableColumn(mockProps)
+      expect(result).toHaveLength(5)
+    })
+  })
+
+  describe('AdminApprovals', () => {
+    const mockState = {
```

Comment:
There is a stub for app state in the reducers folder - can we extend that please just so each test file doesn't have a copy-pasted state object in it. Will make tests more maintainable when we change the state object